torch/nested/_internal/ops.py

```diff
@@ -502,16 +509,76 @@ def zero__default(func, *args, **kwargs):
 
 
 @register_jagged_func(
-    torch.ops.aten._softmax.default, "self: jt, dim: any, half_to_float: any"
+    torch.ops.aten._softmax.default, "self: jt_all, dim: any?, half_to_float: any"
 )
 def _softmax_default(func, *args, **kwargs):
+    """
+    Performs a softmax along the provided tensor dimension.
+    Returns a tensor of the same type as the input.
+    """
     _, new_kwargs = normalize_function(
         func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
     )
 
+    if isinstance(new_kwargs["dim"], tuple):
+        raise RuntimeError(
+            "softmax(): not supported for dimensions of type 'tuple' for NestedTensor"
+        )
+
     inp = new_kwargs.pop("input")
-    dim = new_kwargs["dim"]
-    new_kwargs["dim"] = _wrap_jagged_dim(len(inp._size), dim, "softmax")
+
+    (
+        new_kwargs["dim"],
+        reduce_on_batch,
+        reduce_on_ragged,
+        reduce_on_non_batch,
+    ) = _wrap_jagged_dims(
+        inp.dim(),
+        (new_kwargs["dim"],),
+        "softmax",
+        inp._ragged_idx,
+        allow_batch_dim=True,
```

Comment:
I don't think this is correct. We should probably error if we try to reduce on dim=0.

To check this, I think you can do something like this:
```
x = torch.randn(4, 4, 4)
nt = torch.nested.as_nested_tensor(x, layout=torch.jagged)
dense_softmax = torch.nn.functional.softmax(x, dim=0)
nt_softmax = torch.nn.functional.softmax(nt, dim=0)
assert torch.allclose(dense_softmax, nt_softmax.values())
```

Reply:
noted, and added a test/assert for this!